growthbook/growthbook
# Add support for Experiment CRUD events in webhooks

### Features and Changes

It will be easier to review this PR by commits since the bulk of the changes are reverting the recent reverted PR, then making changes required to ensure the Mongo serialization issue no longer occurs. Also changing of some docs.

- Reverts the recent revert. There was an issue with storing the `ApiFeature` in Mongo because the conditions start with `$` which is not compatible with Mongo. This stringifies it.
- Improves the generated documentation to add schemas under a new section Models
- Groups all existing pages under Endpoints

One of the big changes is an intermediary script that's added to programmatically add to `openapi.yaml` so we don't have to manually add all the schema. It also adds headings so that it's a bit more organized since adding models alongside endpoints can be confusing to readers.

### Dependencies

n/a

### Testing

- Edit a feature that uses condition rules
- Ensure it saves in Mongo successfully

## Screenshots

### Events page

http://localhost:3000/events

Stringified definition can be seen in the below screenshot.

<img width="1291" alt="image" src="https://user-images.githubusercontent.com/113377031/224860666-43500767-cd65-41e3-81d8-3689b7a31992.png">

### Docs

The screenshot below shows changes to the documentation:

1. Groups all tags under "Endpoints"
2. Adds new section "Models" and programmatically adds all new schema definitions there.
3. Links to the `FeatureDefinition` to help with that `FeatureEnvironment#definition` is now a stringified version of the definition.

![CleanShot 2023-03-13 at 17 44 15@2x](https://user-images.githubusercontent.com/113377031/224863549-36582945-c3fa-4584-b13d-f2f15acdce2e.png)
